samples/extensions/graphics_pipeline_library/README.md

```diff
@@ -0,0 +1,140 @@
+pipeline_library_create_info.pVertexInputState   = &vertex_input_state;
+
+vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &pipeline_library_create_info, nullptr, &pipeline_library.vertex_input_interface);
+```
+
+## Deprecating shader modules
+
+With this extension, creating shader modules with `vkCreateShaderModule` has been deprecated and you can instead just pass the shader module create info via `pNext` into your pipeline shader stage create info. This change bypasses a useless copy and is recommended:
+
+```cpp
+VkShaderModuleCreateInfo shader_module_create_info{};
+shader_module_create_info.sType    = VK_STRUCTURE_TYPE_SHADER_MODULE_CREATE_INFO;
+shader_module_create_info.codeSize = static_cast<uint32_t>(spirv.size()) * sizeof(uint32_t);
+shader_module_create_info.pCode    = spirv.data();
+
+VkPipelineShaderStageCreateInfo shader_Stage_create_info{};
+shader_Stage_create_info.sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO;
+// Chain the shader module create info
+shader_Stage_create_info.pNext = &shader_module_create_info;
+shader_Stage_create_info.stage = VK_SHADER_STAGE_VERTEX_BIT;
+shader_Stage_create_info.pName = "main";
+
+VkGraphicsPipelineCreateInfo pipeline_library_create_info{};
+pipeline_library_create_info.stageCount = 1;
+pipeline_library_create_info.pStages    = &shader_Stage_create_info;
+```
+
+You can see this in the pre-rasterization and fragment shader library setup parts of the sample.
+
+## Linking executables
+
+Once all pipeline (library) parts have been created, the pipeline executable can be linked together from them:
+
+```cpp
+std::vector<VkPipeline> libraries = {
+    pipeline_library.vertex_input_interface,
+    pipeline_library.pre_rasterization_shaders,
+    fragment_shader,
+    pipeline_library.fragment_output_interface
+};
+
+// Link the library parts into a graphics pipeline
+VkPipelineLibraryCreateInfoKHR linking_info{};
+linking_info.sType        = VK_STRUCTURE_TYPE_PIPELINE_LIBRARY_CREATE_INFO_KHR;
+linking_info.libraryCount = static_cast<uint32_t>(libraries.size());
+linking_info.pLibraries   = libraries.data();
+
+VkGraphicsPipelineCreateInfo executable_pipeline_create_info{};
+executable_pipeline_create_info.sType = VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO;
+executable_pipeline_create_info.pNext = &linking_info;
+executable_pipeline_create_info.flags = VK_PIPELINE_CREATE_LINK_TIME_OPTIMIZATION_BIT_EXT;
+
+VkPipeline executable = VK_NULL_HANDLE;
+vkCreateGraphicsPipelines(get_device().get_handle(), thread_pipeline_cache, 1, &executable_pipeline_create_info, nullptr, &executable);
+```
+
+This will result in the pipeline state object to be used at draw time. 
+
+A note on `VK_PIPELINE_CREATE_LINK_TIME_OPTIMIZATION_BIT_EXT`: This is an optimization flag. If specified, implementations are allowed to do additional optimization passes. This may increase build times but can in turn result in lower runtime costs.
```

Comment:
I'd suggest it's worth mentioning here that we suggest apps build first without this flag, and then build again with this flag enabled on the slow path, and switch it out for the fast-linked version when available.

Reply:
I also wonder if it's worth making the sample behave like this? The pieces to do it are all there, so I don't think it'd be a big change.

Reply:
That's probably a reasonable compromise.